src/TrackArtist.cpp

```diff
@@ -2114,22 +2113,23 @@ void TrackArtist::DrawClipSpectrum(WaveTrackCache &waveTrackCache,
    enum { DASH_LENGTH = 10 /* pixels */ };
 
    const ClipParameters params(true, track, clip, rect, selectedRegion, zoomInfo);
-   const wxRect &hiddenMid = params.hiddenMid;
+   const wxRect hiddenMid = params.hiddenMid;
+
    // The "hiddenMid" rect contains the part of the display actually
    // containing the waveform, as it appears without the fisheye.  If it's empty, we're done.
    if (hiddenMid.width <= 0) {
       return;
    }
 
-   const double &t0 = params.t0;
-   const double &tOffset = params.tOffset;
-   const double &ssel0 = params.ssel0;
-   const double &ssel1 = params.ssel1;
-   const double &averagePixelsPerSample = params.averagePixelsPerSample;
-   const double &rate = params.rate;
-   const double &hiddenLeftOffset = params.hiddenLeftOffset;
-   const double &leftOffset = params.leftOffset;
-   const wxRect &mid = params.mid;
+   const double t0 = params.t0;
+   const double tOffset = params.tOffset;
+   const double ssel0 = params.ssel0;
+   const double ssel1 = params.ssel1;
+   const double averagePixelsPerSample = params.averagePixelsPerSample;
+   const double rate = params.rate;
+   const double hiddenLeftOffset = params.hiddenLeftOffset;
+   const double leftOffset = params.leftOffset;
+   const wxRect mid = params.mid;
```

Comment:
[Here you go](https://gist.github.com/walisser/bdf50028d6e5e92c2aa3389639ef83fb#file-diff-asm-L1231). The refs version is larger (both stack size and code size).